src/Azure.Functions.Cli/Actions/LocalActions/InstallExtensionAction.cs

```diff
@@ -128,13 +134,62 @@ public async override Task RunAsync()
             }
             else
             {
-                ColoredConsole.Error.WriteLine(ErrorColor(Constants.Errors.ExtensionsNeedDotnet));
+                throw new CliException(Constants.Errors.ExtensionsNeedDotnet);
             }
         }
 
-        private async Task AddPackage(string extensionsProj, string pacakgeName, string version)
+        private bool NeedsExtensionsInstall()
+        {
+            string warningMessage = "No action performed because no functions in your app require extensions.";
+
+            var anyExtensionsToBeInstalled = ExtensionsHelper.GetExtensionPackages().Count() > 0;
+            var extensionsProjDir = string.IsNullOrEmpty(ConfigPath) ? Environment.CurrentDirectory : ConfigPath;
+            var extensionsProjFile = Path.Combine(extensionsProjDir, Constants.ExtenstionsCsProjFile);
+
+            // CASE 1: No extensions.csproj, only need to continue if there are extensions to install
+            if (!FileSystemHelpers.FileExists(extensionsProjFile))
+            {
+                if (!anyExtensionsToBeInstalled)
+                {
+                    if (_showNoActionWarning)
+                    {
+                        ColoredConsole.WriteLine(WarningColor(warningMessage));
+                    }
+                    return false;
+                }
+                return true;
+            }
+
+            // CASE 2: extensions.csproj present with only ExtensionsMetaDataGenerator in it
+            // We look for this special case because we had added ExtensionsMetaDataGenerator to all function apps.
+            // These apps do not need to do a restore, so if only ExtensionsMetaDataGenerator is present, we don't need to continue
+            var extensionsProject = ProjectHelpers.GetProject(extensionsProjFile);
+            var extensionsInProject = extensionsProject.Items
+                    .Where(item => item.ItemType.Equals(Constants.PackageReferenceElementName, StringComparison.OrdinalIgnoreCase))
+                    .ToList();
+            if (extensionsInProject.Count == 1 &&
+                extensionsInProject.FirstOrDefault(item =>
+                item.Include.Equals(Constants.ExtensionsMetadataGeneratorPackage.Name, StringComparison.OrdinalIgnoreCase)) != null)
```

Comment:
You need to check for `!anyExtensionsToBeInstalled` here as well.

Reply:
Ah, good catch. I think I am just going to return true if `anyExtensionsToBeInstalled`